meshio/nastran/_nastran.py

```diff
@@ -192,7 +192,24 @@ def add_cell(cell, cell_type, cell_ref, keyword_prev):
     return mesh
 
 
-def write(filename, mesh):
+# There are two basic categories of input data formats in NX Nastran:
+#
+#     "Free" format data, in which the data fields are simply separated by commas. This type of data is known as free field data.
+#     "Fixed" format data, in which your data must be aligned in columns of specific width. There are two subcategories of fixed format data that differ based on the size of the fixed column width:
+#         Small field format, in which a single line of data is divided into 10 fields that can contain eight characters each.
+#         Large field format, in which a single line of input is expanded into two lines The first and last fields on each line are eight columns wide, while the intermediate fields are sixteen columns wide. The large field format is useful when you need greater numerical accuracy.
+#
+# See: http://help.autodesk.com/view/NSTRN/2019/ENU/?guid=GUID-42B54ACB-FBE3-47CA-B8FE-475E7AD91A00:
+
+
+def write(filename, mesh, format="free"):
+    if format == "free":
+        is_free_format = True
+    elif format == "fixed":
```

Comment:
I would like to see `fixed-small` and `fixed-large` here.

Reply:
"Large" and "small" are not different formats but two possible ways of storing data in "fixed" field format. Tetrahedrons are stored using "small" record while hexahedral cells require "large" variant, because (key + cell_id + mat_id + 8nodes) > 10 fields.

Reply:
Mesh coordinates can be exported in the "short" variant, but the accuracy is limited to 8 characters. So it is better to use the "large" format where 16 characters per number are available (implemented in this PR).

Reply:
I don't understand the format. The spec says that, for using large, you'll have to use two lines. Correct? And the first and last entry in each line can be eight characters. Why?? It seems to defeat the purpose of greater accuracy if only _some_ data is more accurate. What is `mat_id`?

Reply:
By `mat_id`I mean a material id or a cell group.

Yes, the "fixed" format can be a bit confusing. Each line of a file has 80 characters divided into ten blocks of 8 characters.

- short format: item (cell, coordinates) per line, max 10 fields, 8chars per field 

  example: quad cell - key, cell_id, cell group, ND1, ND2, ND3, ND4
  ```
  CQUAD4  18      1       20      21      30      29
  ```

- long format: item = at least two lines, the last field (8 chars) in line is a "continuation tag", which is equal to the first field in the following line

  example: hexahedron
  line1: key, cell id, cell group (empty), ND1, ND2, ND3, ND4, ND4, ND5, ND6, unique continuation tag
  line2: continuation tag, ND7, ND8
  ```
  CHEXA   1000            1331    449     241     341     159     123     +13e8   
  +13e8   122     141
  ```

Coordinates - long format means 2 * 8chars per coordinate, two lines:
line1: key (`GRID*` = long grid), id, node group (empty), X, Y
line2: `*` = continuation, Z
```
GRID*   45                                  6.000000E-02    3.000000E-02
*           0.000000E+000
```               

Short format:
key, id, node group (empty), X=0.003456, Y=0.001234, Z=0.005678
```
GRID    45      0.0034560.0012340.005678
```

Reply:
Things I don't understand:
 * Where are the 16 digits in the long format for cells?
 * How is the last entry in long nodes `0.000000E+000`. According to the spec, we only have 8 digits.

Reply:
Ah, I get the second question: Each line is divided into a grid if 8+16+16+16+8 columns. The first and the last are not the acutal entries, but the columns.

Reply:
- As far as I know, there are no 16 digits for cells.
- Right, 8+16+16+16+16+8 is the line division for long nodes. The keywords and continuation tags are always limited to 8 characters.

Reply:
> As far as I know, there are no 16 digits for cells. 

But the documentation says so, doesn't it? If I understand correctly, this
```
CHEXA   1000            1331    449     241     341     159     123     +13e8   
+13e8   122     141
```
shouldn't exist. (Sorry if I'm getting into much detail here, I just want to make sure we don't implement our own standard.)

Reply:
> But the documentation says so, doesn't it? 

There is no specification for 16 digit cell entries in [1] (I did find any, but, of cource, I have not read all the pages).

> If I understand correctly, this

> ```
> CHEXA   1000            1331    449     241     341     159     123     +13e8   
> +13e8   122     141
> ```
> 
> shouldn't exist.

This is according to the documentation, see specifications around page 30 and examples at pages 32, 167, 747, 865.
The continuation tag can be an arbitrary alphanumerical string, see page 31. I use a combination of extension line number and hexadecimal value of cell id (1. extra line + 3e8 --> +13e8).

> (Sorry if I'm getting into much detail here, I just want to make sure we don't implement our own standard.)

That is fine, I understand to your approach.